In [22]:
import numpy as np
import pandas as pd

In [23]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [24]:
# Observed data in a (hypothetical) survey of 300 people 
observed = pd.Series([10.83, 8.32, 8.66, 10.55, 7.68, 8.15], index=["al","ar","la","ms","ok","tx"])

In [25]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [26]:
# Add a column whose default values are the expected values
df[1] = 8.68

In [27]:
# Rename columns
df.columns = ["observed", "expected"]

In [28]:
# View the data frame
df

observed  expected
al     10.83      8.68
ar      8.32      8.68
la      8.66      8.68
ms     10.55      8.68
ok      7.68      8.68
tx      8.15      8.68

In [29]:
# The degree of freedom is 3-1 = 2
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 5)

In [30]:
# The critical value
critical_value

11.070497693516351

In [31]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=1.097960829493088, pvalue=0.9542840137912635)

### Conclusion

* Since the chi square value of 1.10 does not exceed the critical value of 11.07, we conclude that the results are not statistically significant.